core/trino-main/src/main/java/io/trino/operator/OrderByOperator.java

```diff
@@ -286,11 +285,7 @@ public Page getOutput()
             return null;
         }
         Page nextPage = next.get();
-        Block[] blocks = new Block[outputChannels.length];
-        for (int i = 0; i < outputChannels.length; i++) {
-            blocks[i] = nextPage.getBlock(outputChannels[i]);
-        }
-        return new Page(nextPage.getPositionCount(), blocks);
+        return nextPage.getColumns(outputChannels);
```

Comment:
`getColumns()` calls `wrapBlocksWithoutCopy(positionCount, blocks)`, while `new Page(count, blocks)` uses `blocksCopyRequired = true`. Have you analyzed the impact of that change in semantics?

Reply:
Yep, that's the intention behind this change. The `Block[]` array in the previous implementation was created locally (not shared anywhere else) but was created once and then immediately cloned by the page constructor unnecessarily. Now we only create a single copy of the blocks array inside of `Page#getColumns` which requires no additional copy (and makes the code a little cleaner).